plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py

```diff
@@ -404,3 +405,190 @@ def generate_tip(total_bill: float, index: int) -> float:
+    def create_years(year: int, reps: int) -> typing.List[int]:
+        return [year + i for i in range(reps)]
+
+    def create_empty(val: typing.Any, reps: int) -> typing.List[typing.Any]:
+        return [val, *[np.nan for _ in range(reps - 1)]]
+
+    # split gapminder into pre-2007 and 2007, since 2007 need not be expanded in the same way as previous years
+    gapminder_no_2007 = typing.cast(pd.DataFrame, gapminder[gapminder["year"] != 2007])
+    gapminder_2007 = typing.cast(pd.DataFrame, gapminder[gapminder["year"] == 2007])
+
+    # expand pre-2007 dataset into consecutive years, where each new year gets filled with np.nan
+    gapminder_no_2007.loc[:, ["year"]] = gapminder_no_2007["year"].apply(
+        lambda x: create_years(x, 5)
+    )
+    gapminder_no_2007.loc[:, ["lifeExp", "pop", "gdpPercap"]] = gapminder_no_2007[
+        ["lifeExp", "pop", "gdpPercap"]
+    ].map(lambda x: create_empty(x, 5))
+    gapminder_no_2007 = gapminder_no_2007.explode(
+        column=["year", "lifeExp", "pop", "gdpPercap"]
+    )
+
+    # insert month column, where each element is a list of all 12 months
+    gapminder_no_2007.insert(
+        loc=3,
+        column="month",
+        value=typing.cast(
+            pd.Series,
+            [[month for month in range(1, 13)] for _ in range(len(gapminder_no_2007))],
+        ),
+    )
+
+    # expand pre-2007 dataset into consecutive months
+    gapminder_no_2007.loc[:, ["lifeExp", "pop", "gdpPercap"]] = gapminder_no_2007[
+        ["lifeExp", "pop", "gdpPercap"]
+    ].map(lambda x: create_empty(x, 12))
+    gapminder_no_2007 = gapminder_no_2007.explode(
+        column=["month", "lifeExp", "pop", "gdpPercap"]
+    )
+
+    # insert a month column into the 2007 dataset to merge the two easily
+    gapminder_2007.insert(
+        loc=3,
+        column="month",
+        value=typing.cast(pd.Series, [1 for _ in range(len(gapminder_2007))]),
+    )
+
+    # combine expanded pre-2007 dataset with 2007 dataset, sort by country and year
+    gapminder_combined = (
+        pd.concat([gapminder_no_2007, gapminder_2007])
+        .sort_values(by=["country", "year"])
+        .reset_index(drop=True)
+    )
+
+    # define types of new combined dataset
+    gapminder_combined = gapminder_combined.astype(
+        {
+            "year": "int",
+            "month": "int",
+            "lifeExp": "float",
+            "pop": "float",
+            "gdpPercap": "float",
+        }
+    )
+
+    # compute linearly interpolated values for each month, round population
+    gapminder_interp_vals = typing.cast(
+        pd.DataFrame,
+        (
+            gapminder_combined.groupby("country", observed=True, as_index=True)[
+                ["lifeExp", "pop", "gdpPercap"]
+            ]
+            .apply(lambda country: country.interpolate(method="linear"))
+            .reset_index(drop=True)
+        ),
+    )
+    gapminder_interp_vals.loc[:, "pop"] = gapminder_interp_vals["pop"].apply(
+        lambda x: round(x)
```

Comment:
`lambda x: round(x)` is equivalent to just passing round in this case
```suggestion
        round
```